src/test/kotlin/no/nav/syfo/domain/inntektsmelding/InntektsmeldingTest.kt

```diff
@@ -70,4 +72,26 @@ internal class InntektsmeldingTest {
         assertFalse(im1.copy(bruttoUtbetalt = BigDecimal(123)).isDuplicate(im1))
         assertFalse(im1.copy(årsakEndring = "qwe").isDuplicate(im1))
     }
+
+    @Test
+    fun `selvbestemt matcher spleis bare hvis vedtaksperiodeId er satt`() {
+        val selvbestemtUtenVedtaksperiodeId = im1.copy(vedtaksperiodeId = null, avsenderSystem = AvsenderSystem(Avsender.NAV_NO_SELVBESTEMT))
+        val selvbestemtMedVedtaksperiodeId = im1.copy(vedtaksperiodeId = UUID.randomUUID(), avsenderSystem = AvsenderSystem(Avsender.NAV_NO_SELVBESTEMT))
+        assertFalse(selvbestemtUtenVedtaksperiodeId.matcherSpleis())
+        assertTrue(selvbestemtMedVedtaksperiodeId.matcherSpleis())
+    }
+
+    @Test
+    fun `Alle avsendere bortsett fra selvbestemt matcher alltid spleis som default`() {
+        val imFraSimbaUtenVedtaksperiodeId = im1.copy(vedtaksperiodeId = null, avsenderSystem = AvsenderSystem(Avsender.NAV_NO))
+        val imFraSimbaMedVedtaksperiodeId = im1.copy(vedtaksperiodeId = UUID.randomUUID(), avsenderSystem = AvsenderSystem(Avsender.NAV_NO))
+        val imFraEksterntSystemUtenVedtaksperiodeId = im1.copy(vedtaksperiodeId = null, avsenderSystem = AvsenderSystem("Whatever Payment Co"))
+        val imFraEksterntSystemMedVedtaksperiodeId = im1.copy(vedtaksperiodeId = UUID.randomUUID(), avsenderSystem = AvsenderSystem("Donald Duck & Co"))
+        val selvbestemtMedVedtaksperiodeId = im1.copy(vedtaksperiodeId = null, avsenderSystem = AvsenderSystem(Avsender.NAV_NO_SELVBESTEMT))
```

Comment:
Denne skal vel kanskje hete selvbestemt*Uten*VedtaksperiodeId ? 😄

Reply:
Bare en test for å sjekke at du følger med. Bra spotta!